src/components/RandomProductsBanner/RandomProductsBanner.js

```diff
@@ -1,9 +1,136 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
 
+import TestApiService from '../../utils/test-api-service';
+import LoadingSpinner from '../LoadingSpinner';
+
+import messages from './messages';
+import { randomProductsBannerOptions } from './constants';
+
+import './RandomProductsBanner.css';
+
+const { updateInterval, cntRandomItems } = randomProductsBannerOptions;
 class RandomProductsBanner extends React.PureComponent {
+  static defaultProps = {
+    updateInterval,
+    cntRandomItems,
+  };
+
+  static propTypes = {
+    updateInterval: PropTypes.number,
+    cntRandomItems: PropTypes.number,
+  };
+
+  testswapiService = new TestApiService();
+
+  state = {
+    products: {},
+    loading: true,
+    error: false,
+  };
+
+  componentDidMount() {
+    const { updateInterval } = this.props;
+    this.updateBanner();
+    this.interval = setInterval(this.updateBanner, updateInterval);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  onBannerLoaded = products => {
+    this.setState({
+      products,
+      loading: false,
+    });
+  };
+
+  onError = () => {
+    this.setState({
+      error: true,
+      loading: false,
+    });
+  };
+
+  updateBanner = () => {
+    const { cntRandomItems } = this.props;
+
+    this.testswapiService
+      .getRandomProducts(cntRandomItems)
+      .then(this.onBannerLoaded)
+      .catch(this.onError);
+  };
+
   render() {
-    return <div>Random Products Banner</div>;
+    const { products, loading, error } = this.state;
+
+    const hasData = !(loading || error);
+
+    const errorMessage = error ? <Error /> : null;
+    const spinner = loading ? <LoadingSpinner /> : null;
+    const content = hasData ? <BannerView products={products} /> : null;
+
+    return (
+      <div className="random-products-banner">
+        {errorMessage}
+        {spinner}
+        {content}
+      </div>
+    );
   }
 }
 
+const getProductItem = ({ id, name, img, description, price, type }) => {
```

Comment:
let's keep to the same code style and keep functions and components above the class functions that use them. So please move it above the class Banner.